File: 백준/Silver/15650. N과 M （2）/N과 M （2）.py
n, m = map(int, input().split())

arr = list(map(str, range(1, n + 1)))

def recursive(li, k):
    if k == m:
        print(" ".join(li))
        return

    for i in arr:
        if not li or li[-1] < i:
            li.append(i)
            recursive(li, k + 1)
            li.pop()


recursive([], 0)

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.js
const fs = require('fs');
const [n, k] = fs.readFileSync('/dev/stdin').toString().trim().split(" ").map(Number);
const li = [];
const recur = () => {
    if (li.length == k) {
        console.log(li.join(" "));
        return;
    }
    
    for(let i = 1; i <= n; i++){
        if (li.length == 0 || i > li.at(-1)){
            li.push(i)
            recur()
            li.pop()    
        }
    }
}

recur()